### PR TITLE
Feat/sam3 integration refactor

### LINKED ISSUE #5: another refactor!
tidy up stale scripts and create a single experiment launcher - now those are spread out accross different files

also, I think we dont really need to have all the current expeirments - they can be well redesigned

### LINKED ISSUE #6: docs
move docs into docs

add also more docs about current work

### LINKED ISSUE #14: caching fix
rerun experiments with no cache logic
clean the code, so there is no caching logic (for now, no code = no confusion)

### PR DESCRIPTION
**Pull request by shpirm**
Closes #6 #5

### CODE REVIEW

File: idisc/models/sam3_video_encoder.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+                        resized.append(feat)
+                    per_frame_fpn[frame_idx] = resized
+
+                # 2. Capture queries from hook (fired during detector
+                #    forward inside propagate_in_video).
+                if self._last_hook_hs is not None:
+                    hs = self._last_hook_hs
+                    k = min(self.top_k_queries, hs.shape[0])
+                    norms = hs.float().norm(dim=-1)
+                    _, top_idx = norms.topk(k)
```
Comment (Fersoil) [inline idisc/models/sam3_video_encoder.py]: why do we calculate here topk based on the norms? not queries as in the image encoder?

File: scripts/experiments/finetune.py [hunk 2]
```diff
@@ -0,0 +1,645 @@
...
+                        with context:
+                            pred, losses, _ = model(
+                                images[b, t : t + 1],  # still needed for shape
+                                instance_queries=iq,
+                                sam_mode=sam_mode,
+                                pre_extracted_encoder_outputs=inv_frame_fpn,
+                                gt=gt_t,
+                                mask=mask_t,
+                            )
+                            loss = sum(v for v in losses["opt"].values()) / n_samples
```
Comment (Fersoil) [inline scripts/experiments/finetune.py]: I think there it could be a problem with total loss, total_loss below should be:
```
total_loss += loss.item() * n_samples
```

Comment (Fersoil) [reply scripts/experiments/finetune.py]: or just maybe multiply by T

**Review by Fersoil — APPROVED** [review]
looks good to merge

**Review by Fersoil — APPROVED** [review]

File: idisc/models/sam3_video_encoder.py (hunk 1 above)
Comment (shpirm) [reply idisc/models/sam3_video_encoder.py]: topk code should be cleaned from everywhere, no need in this
